app/survey_config/business_config.py

```diff
@@ -55,10 +56,25 @@ def __post_init__(self):
         ]
 
     def get_service_links(
-        self, sign_out_url: str, *, is_authenticated: bool
+        self, sign_out_url: str, *, is_authenticated: bool, ru_ref: Optional[str]
     ) -> Optional[list[dict]]:
+
+        if self.schema and is_authenticated and ru_ref:
+            survey_id = self.schema.json["survey_id"]
+            self.account_service_help_url = f"{ACCOUNT_SERVICE_BASE_URL}/surveys/surveys-help?survey_ref={survey_id}&ru_ref={ru_ref}"
+
+        else:
+            self.account_service_help_url = f"{ACCOUNT_SERVICE_BASE_URL}/help"
```

Comment:
Just realised this is using the constant `ACCOUNT_SERVICE_BASE_URL`, it should be using `self.base_url`. I think we should actually move this out into its own method and remove the object property `self.account_service_help_url`. The return statement could be simplified since we will always have 1 link (help).


Something like:

```py
    def _get_account_service_help_url(
        self, *, is_authenticated: bool, ru_ref: Optional[str]
    ) -> str:
        if self.schema and is_authenticated and ru_ref:
            survey_id = self.schema.json["survey_id"]
            return f"{self.base_url}/surveys/surveys-help?survey_ref={survey_id}&ru_ref={ru_ref}"

        return f"{self.base_url}/help"

    def get_service_links(
        self, sign_out_url: str, *, is_authenticated: bool, ru_ref: Optional[str]
    ) -> Optional[list[dict]]:
        links = [
            HeaderLink(
                lazy_gettext("Help"),
                self._get_account_service_help_url(
                    is_authenticated=is_authenticated, ru_ref=ru_ref
                ),
                id="header-link-help",
            ).__dict__
        ]

        if is_authenticated:
            links.extend(
                [
                    HeaderLink(
                        lazy_gettext("My account"),
                        self.account_service_my_account_url,
                        id="header-link-my-account",
                    ).__dict__,
                    HeaderLink(
                        lazy_gettext("Sign out"),
                        sign_out_url,
                        id="header-link-sign-out",
                    ).__dict__,
                ]
            )

        return links
```

Reply:
Added this, thanks! Had to modify some tests since we now use base url but it works fine now.